docs/src/pages/guides/migration-v4/migration-v4.md

```diff
@@ -2501,3 +2501,85 @@ declare module "@material-ui/private-theming" {
   interface DefaultTheme extends Theme {}
 }
 ```
+
+### makeStyles - TypeError: Cannot read property 'drawer' of undefined
+
+This error occurs when calling `useStyles` (result of `makeStyles`) or `withStyles` outside of `<ThemeProvider>` scope like this:
+
+```js
+import * as React from 'react';
+import { ThemeProvider, createTheme } from '@material-ui/core/styles';
+import makeStyles from '@material-ui/styles/makeStyles';
+import Card from '@material-ui/core/Card';
+import CssBaseline from '@material-ui/core/CssBaseline';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    display: 'flex',
+    backgroundColor: theme.palette.primary.main,
+    color: theme.palette.common.white,
+  },
+}));
+
+const theme = createTheme();
+
+function App() {
+  const classes = useStyles(); // ❌ called outside of ThemeProvider
+  return (
+    <ThemeProvider theme={theme}>
+      <CssBaseline />
+      <Card className={classes.root}>...</Card>
+    </ThemeProvider>
+  );
+}
+
+export default App;
+```
+You can fix by creating **AppContent** component or **withThemeProvider** HOC (higher order component)
+
+**Option 1 : `AppContent`**
+
+```js
+// ...imports
+
+function AppContent(props) {
+  const classes = useStyles(); // ✅ This is safe because AppContent is rendered inside <ThemeProvider>
+  return <Card className={classes.root}>...</Card>;
+}
+
+const App = (props) => {
+  return (
+    <ThemeProvider theme={theme}>
+      <CssBaseline />
+      <AppContent {...props} />
+    </ThemeProvider>
+  );
+};
+
+export default App;
+```
+
+**Option 2 : `withThemeProvider`**
+
+```js
+// ...imports
+
+function App() {
+  const classes = useStyles(); // ✅ This is safe because App is now rendered inside <ThemeProvider>
+  return <Card className={classes.root}>...</Card>;
+}
+
+const withThemeProvider = (Component) => {
```

Comment:
The component created is used only once in your entire app. It isn't justified to use higher-order components for this use case.

Otherwise please elaborate on the tradeoffs between these two options. It's really important to understand that more options aren't free. An API isn't better because it has more options for the same behavior. Quite the opposite: An API is strictly worse if it has multiple options for the same behavior because we don't solve any more problems but double the maintenance burden and double the choices a user has to make.

Reply:
I am fine to remove `withThemeProvider` option and only suggest `AppContent` option in the docs.

Reply:
> I am fine to remove `withThemeProvider` option and only suggest `AppContent` option in the docs.

+1 on this